sklearn/metrics/_classification.py

```diff
@@ -1638,47 +1693,32 @@ def precision_recall_fscore_support(
         if (pred_sum[true_sum == 0] == 0).any():
             _warn_prf(average, "true nor predicted", "F-score is", len(true_sum))
 
-    # if tp == 0 F will be 1 only if all predictions are zero, all labels are
-    # zero, and zero_division=1. In all other case, 0
     if np.isposinf(beta):
         f_score = recall
+    elif beta == 0:
+        f_score = precision
     else:
+        # set to zero_division_value if denom 0
+        # OR if BOTH precision and recall are ill-defined
         denom = beta2 * precision + recall
-
-        denom[denom == 0.0] = 1  # avoid division by 0
+        mask = (denom == 0.0) | ((pred_sum + true_sum) == 0)
```

Comment:
```suggestion
        mask = (denom == 0.0) | ((pred_sum + true_sum) == 0.0)
```

I really dislike that we do a pure comparison with floating point.
I am wondering if there is a downside with the following:

```python
mask = np.isclose(denom, 0) | np.isclose(pred_sum + true_sum, 0)
```

Reply:
good call. Passing tests, so will use this